cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/FileLocalizer.java

```diff
@@ -64,17 +65,16 @@ public void run(String[] args) throws Exception {
 
     // Localize the runtime config jar
     URI uri = URI.create(args[0]);
-    Location runtimeConfigLocation;
-    if (context.getLocationFactory().getHomeLocation().toURI().getScheme().equals(uri.getScheme())) {
-      runtimeConfigLocation = context.getLocationFactory().create(uri);
-    } else {
-      runtimeConfigLocation = new LocalLocationFactory().create(new File(uri).toURI());
-    }
+    Location runtimeConfigLocation = new LocalLocationFactory().create(new File(uri.getPath()).toURI());
+
+    masterEnv.fetchFile(uri, new File(runtimeConfigLocation.toURI().getPath()));
```

Comment:
Shouldn't the fetch method should be in the context instead of environment? The `MasterEnvironmentContext` is for CDAP to provide core services to the `MasterEnvironment` implementations. The `MasterEnvironment` is for CDAP to call into the extension implementation.

Reply:
As mentioned above, initiated tried to do in context, but context created before master env creation, where discovery service gets created.